libs/partners/google-vertexai/langchain_google_vertexai/chat_models.py

```diff
@@ -156,6 +173,16 @@ def _convert_to_prompt(part: Union[str, Dict]) -> Part:
         if isinstance(raw_content, str):
             raw_content = [raw_content]
         parts = [_convert_to_prompt(part) for part in raw_content]
+
+        if raw_system_message:
```

Comment:
given this logic only applies to the beginning of the loop, it'd be cleaner to handle this before the loop! Any chance you could change that for readability?

Reply:
There is a bit more dependency and would make the code redundant if the logic is pulled out of the loop. Also this is a conditional logic that gets applied to element=1 when element=0 meets a condition.